### PR TITLE
fix(core): only warn and auto declare undeclared `entryComponents`.

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/@angular/core/test/linker/ng_module_integration_spec.ts [hunk 1]
```diff
@@ -281,6 +271,21 @@ function declareTests({useJit}: {useJit: boolean}) {
...
+           @NgModule({entryComponents: [SomeComp]})
+           class SomeModule {
+           }
+
+           const ngModule = createModule(SomeModule);
+           expect(ngModule.componentFactoryResolver.resolveComponentFactory(SomeComp).componentType)
+               .toBe(SomeComp);
+
+           expect(console.warnings).toEqual([
+             `NgModule ${stringify(SomeModule)} uses ${stringify(SomeComp)} via "entryComponents" but it was neither declared nor imported! This warning will become an error after final.`
```
Comment (IgorMinar) [inline modules/@angular/core/test/linker/ng_module_integration_spec.ts]: we need to make this error message more actionable.

we should actually check if the module name is DynamicModule and if it then we should say that people should switch over to modules. If the name is different then we should tell people to add the components to declarations.

this made me realize that if we do want to be strict about this in the future then we can't have DynamicModule and simple bootstrap. :-/

**Comment by IgorMinar** [comment]
this change looks good but let's discuss it with others before merging it

**Comment by IgorMinar** [comment]
Upon further discussion we decided to go forward with this change.

File: modules/@angular/core/test/linker/ng_module_integration_spec.ts (hunk 1 above)
Comment (tbosch) [inline modules/@angular/core/test/linker/ng_module_integration_spec.ts]: Didn't make this change of the error message for now...

**Pull request closed without merging**